### ISSUE TITLE
Possible improvements to in operator

### ISSUE THREAD

**Issue by pablojim** [dask/fastparquet#400]
The current "in" operator filter implementation can return more row groups than necessary from the existing row group statistics.

e.g:
```
df = pd.DataFrame({'a': range(100),'b': np.random.rand(100)})
write(parquet_path, df, row_group_offsets=10)

res = list(pf.iter_row_groups(filters=[('a', 'in', [9])]))
len(res) # 1 group

res = list(pf.iter_row_groups(filters=[('a', 'in', [99])]))
len(res) # 1 group

res = list(pf.iter_row_groups(filters=[('a', 'in', [9, 99])]))
len(res) # 10 groups returned
```

This is due to max/min handling here: https://github.com/dask/fastparquet/blob/master/fastparquet/api.py#L795

Is this desired behaviour? maybe for performance reasons?

Would a PR that instead considers each value in the set separately be considered?

**Comment by martindurant** [comment]
Yes, I believe you are right, and the last line should return just the two groups. A PR would certainly be welcome.

**Comment by pablojim** [comment]
closed by #401
